src/accessorysetupkit.cs

```diff
@@ -0,0 +1,208 @@
+		AwaitingAuthorization   = 10,
+		Authorized              = 20,
+	}
+
+	[Flags]
+	[Native]
+	[iOS (18, 0)]
+	public enum ASAccessoryRenameOptions : ulong {
+	    Ssid = 1U << 0,
+	}
+
+	[Flags]
+	[Native]
+	[iOS (18, 0)]
+	public enum ASAccessorySupportOptions : ulong {
+		BluetoothPairingLE = 1U << 1,
+		BluetoothTransportBridging = 1U << 2,
+	}
+
+	[BaseType (typeof (NSObject))]
+	[iOS (18, 0)]
+	[DisableDefaultCtor]
+	interface ASAccessory {
+		[Export ("state", ArgumentSemantic.Assign)]
+		ASAccessoryState State { get; }
+
+		[Export ("bluetoothIdentifier", ArgumentSemantic.Copy), NullAllowed]
+		NSUuid BluetoothIdentifier { get; }
+
+		[Export ("displayName", ArgumentSemantic.Copy)]
+		string DisplayName { get; }
+
+		[Export ("SSID", ArgumentSemantic.Copy), NullAllowed]
+		string Ssid { get; }
+
+		[Export ("descriptor", ArgumentSemantic.Copy)]
+		ASDiscoveryDescriptor Descriptor { get; }
+	}
+
+	[Native]
+	[iOS (18, 0)]
+	public enum ASAccessoryEventType : long {
+	    Unknown           = 0,
+	    Activated         = 10,
+	    Invalidated       = 11,
+	    MigrationComplete = 20,
+	    AccessoryAdded    = 30,
+	    AccessoryRemoved  = 31,
+	    AccessoryChanged  = 32,
+	    PickerDidPresent  = 42,
+	    PickerDidDismiss  = 45,
+	}
+
+	[BaseType (typeof (NSObject))]
+	[iOS (18, 0)]
+	[DisableDefaultCtor]
+	interface ASAccessoryEvent {
+		[Export ("eventType", ArgumentSemantic.Assign)]
+		ASAccessoryEventType EventType { get; }
+
+		[Export ("accessory", ArgumentSemantic.Copy), NullAllowed]
+		ASAccessory Accessory { get; }
+
+		[Export ("error", ArgumentSemantic.Copy), NullAllowed]
+		NSError Error { get; }
+	}
+
+	delegate void ASAccessorySessionCompletionHandler ([NullAllowed] NSError error);
+
+	[BaseType (typeof (NSObject))]
+	[iOS (18, 0)]
+	[DisableDefaultCtor]
+	interface ASAccessorySession {
+		[Export ("accessories", ArgumentSemantic.Copy)]
+		ASAccessory [] Accessories { get; }
+
+		[Export ("activateWithQueue:eventHandler:")]
+		void Activate (DispatchQueue queue, Action<ASAccessoryEvent> eventHandler);
+
+		[Export ("invalidate")]
+		void Invalidate ();
+
+		[Export ("showPickerWithCompletionHandler:")]
+		void ShowPicker (ASAccessorySessionCompletionHandler completionHandler);
+
+		[Export ("showPickerForDisplayItems:completionHandler:")]
+		void ShowPicker (ASPickerDisplayItem [] displayItems, ASAccessorySessionCompletionHandler completionHandler);
+
+		[Export ("finishAuthorization:settings:completionHandler:")]
+		void FinishAuthorization (ASAccessory accessory, ASAccessorySettings settings, ASAccessorySessionCompletionHandler completionHandler);
+
+		[Export ("removeAccessory:completionHandler:")]
+		void RemoveAccessory (ASAccessory accessory, ASAccessorySessionCompletionHandler completionHandler);
+
+		[Export ("renameAccessory:options:completionHandler:")]
+		void RenameAccessory (ASAccessory accessory, ASAccessoryRenameOptions renameOptions, ASAccessorySessionCompletionHandler completionHandler);
```

Comment:
Any reason for not doing these `[Async]`?

Reply:
Fixed to be [Async].